engine/src/main/java/org/terasology/logic/players/CameraClientSystem.java

```diff
@@ -76,4 +96,11 @@ public void resetCamera(ResetCameraEvent resetCameraEvent, EntityRef client, Aut
         }
         Location.attachChild(targetEntityForCamera, clientComponent.camera, cameraPosition, new Quat4f(Quat4f.IDENTITY));
```

Comment:
I know you haven't touched this code, but I'm wondering if we can perhaps remove the Vector3f cameraPosition declaration and just place two Location.attachChild() calls within the IF statement. Also, is cameraLocation == null the expected typical case? If not I'd recommend switching the condition to cameraLocation != null.